resource/js/vocab-search.js

```diff
@@ -7,18 +7,119 @@ const vocabSearch = Vue.createApp({
       languages: [],
       selectedLanguage: null,
       searchTerm: null,
-      autoCompeteResults: [],
-      languageStrings: null
+      searchCounter: null,
+      renderedResultsList: [],
+      languageStrings: null,
+      msgs: null
     }
   },
   mounted () {
     this.languages = SKOSMOS.languageOrder
     this.selectedLanguage = SKOSMOS.content_lang
-    this.languageStrings = SKOSMOS.language_strings[SKOSMOS.lang]
+    this.searchCounter = 0
+    this.languageStrings = SKOSMOS.language_strings[SKOSMOS.lang] ?? SKOSMOS.language_strings.en
+    this.msgs = SKOSMOS.msgs[SKOSMOS.lang] ?? SKOSMOS.msgs.en
+    this.renderedResultsList = []
+    document.addEventListener('click', this.onClickOutside)
+  },
+  beforeUnmount () {
+    document.removeEventListener('click', this.onClickOutside)
   },
   methods: {
     autoComplete () {
+      const delayMs = 300
+
+      // when new autocomplete is fired, empty the previous result
+      this.hideDropdown()
+
+      // cancel pending API calls when method is called
+      clearTimeout(this._timerId)
+
+      // is the search term is in cache, use the cache
+
+      // delay call, but don't execute if the search term is not at least two characters
+      if (this.searchTerm.length > 1) {
+        this._timerId = setTimeout(() => { this.search() }, delayMs)
+      }
+    },
+    /*
+     * search should fetch the response and save it to cache
+     * search calls renderResults for displaying the response
+     */
+    search () {
+      const mySearchCounter = this.searchCounter + 1 // make sure we can identify this search later in case of several ongoing searches
+      this.searchCounter = mySearchCounter
+
+      fetch('rest/v1/' + SKOSMOS.vocab + '/search?query=' + this.formatSearchTerm() + '&lang=' + SKOSMOS.lang)
+        .then(data => data.json())
+        .then(data => {
+          if (mySearchCounter === this.searchCounter) {
+            this.renderedResultsList = data.results // update results (update cache if it is implemented)
+            this.renderResults() // render after the fetch has finished
+          }
+        })
+    },
+    formatSearchTerm () {
+      if (this.searchTerm.includes('*')) { return this.searchTerm }
+      const formatted = this.searchTerm + '*'
+      return formatted
+    },
+    renderMatchingPart (searchTerm, label) {
+      const regex = new RegExp(searchTerm, 'i') // case insensitive matching
+      if (label && regex.test(searchTerm)) {
+        return label.replace(regex, (match) => `<b>${match}</b>`)
+      }
+      return label
+    },
+    translateType (type) {
+      return SKOSMOS.msgs[SKOSMOS.lang][type]
```

Comment:
This method of displaying human-readable type labels only works in the case where the types are defined in UI translations (for which the infrastructure in Skosmos 3 is still lacking). But Skosmos 2 has another mechanism as well: the frontend will call the `types` REST API method and use its results to display human-readable type labels, which are typically vocabulary-specific types that have `rdfs:label` properties in the vocabulary data.

I don't think this PR needs to use the `types` method or otherwise know how to display vocabulary-specific type labels, but this missing feature should be put on the TODO list if it's not yet there. (Though it doesn't have to be implemented the same way as in Skosmos 3; maybe the type information could be passed to the frontend code already in the SKOSMOS object, because it's going to be needed in most cases anyway. This would reduce the number of REST API calls.)

Reply:
I added this to the relevant issue as "3.5.1 Search result types should be fetched from rest/v1/types and stored in SKOSMOS object".

Reply:
OK good. I edited the wording slightly to this: "Labels for search result types should be provided by the backend in the SKOSMOS object and displayed in search results"
...because the types don't need to be fetched from the REST API if they are already given in the SKOSMOS object.